OpenEdXMobile/src/main/java/org/edx/mobile/util/DataBindingHelperUtils.kt

```diff
@@ -61,16 +63,24 @@ class DataBindingHelperUtils {
                 list.forEach { item ->
                     val childView = inflater.inflate(R.layout.sub_item_course_date_block, null)
 
-                    setText(childView.title, item.title)
-                    isViewAccessible(childView.title, item.dateBlockBadge)
+                    var labelType = ""
+                    val title = SpannableString(item.title)
 
-                    setText(childView.description, item.description)
-                    isViewAccessible(childView.description, item.dateBlockBadge)
+                    if (item.assignmentType.isNullOrBlank().not()) {
+                        labelType = "${item.assignmentType}: "
+                    }
 
                     if (item.showLink()) {
-                        childView.title.paintFlags = Paint.UNDERLINE_TEXT_FLAG
+                        title.setSpan(UnderlineSpan(), 0, title.length, 0)
                     }
 
+                    childView.title.visibility = if (item.title.isNullOrBlank()) View.GONE else View.VISIBLE
+                    childView.title.text = TextUtils.concat(labelType, title)
+                    isViewAccessible(childView.title, item.dateBlockBadge)
```

Comment:
better to use the following line of code.

```
isViewVisible(childView.title, item.title.isNullOrBlank())
setText(childView.title, TextUtils.concat(labelType, title).toString())
isViewAccessible(childView.title, item.dateBlockBadge)
```

Reply:
`TextUtils.concat(labelType, title).toString()` will remove the effect of spanning and simple string will be passed to the function